src/thok.rs

```diff
@@ -151,7 +158,7 @@ impl Thok {
     }
 
     pub fn backspace(&mut self) {
-        if self.cursor_pos > 0 {
+        if self.cursor_pos > 0 && self.cursor_pos > self.total_line_length {
```

Comment:
```suggestion
        if self.cursor_pos > self.total_line_length {
```
I believe you should be good to shorten this